src/systems/altimeter/Altimeter.cc

```diff
@@ -51,30 +52,35 @@ using namespace systems;
 /// \brief Private Altimeter data class.
 class ignition::gazebo::systems::AltimeterPrivate
 {
-  /// \brief A map of altimeter entity to its vertical reference
+  /// \brief A map of altimeter entity to its sensor
   public: std::unordered_map<Entity,
       std::unique_ptr<sensors::AltimeterSensor>> entitySensorMap;
 
   /// \brief Ign-sensors sensor factory for creating sensors
   public: sensors::SensorFactory sensorFactory;
 
+  /// \brief Keep list of sensors that were created during the previous
+  /// `PostUpdate`, so that components can be created during the next
+  /// `PreUpdate`.
+  public: std::unordered_set<Entity> newSensors;
+
   /// True if the rendering component is initialized
   public: bool initialized = false;
 
   /// \brief Create sensor
-  /// \param[in] _ecm Mutable reference to ECM.
+  /// \param[in] _ecm Immutable reference to ECM.
   /// \param[in] _entity Entity of the IMU
   /// \param[in] _altimeter Altimeter component.
   /// \param[in] _parent Parent entity component.
   public: void AddAltimeter(
-    EntityComponentManager &_ecm,
+    const EntityComponentManager &_ecm,
     const Entity _entity,
     const components::Altimeter *_altimeter,
     const components::ParentEntity *_parent);
 
   /// \brief Create altimeter sensor
-  /// \param[in] _ecm Mutable reference to ECM.
-  public: void CreateAltimeterEntities(EntityComponentManager &_ecm);
+  /// \param[in] _ecm Immutable reference to ECM.
+  public: void CreateAltimeterEntities(const EntityComponentManager &_ecm);
```

Comment:
```suggestion
  public: void CreateAltimeterSensors(const EntityComponentManager &_ecm);
```

It looks like this method can be renamed for clarity. Even the docs say this this method should create an altimeter sensor (no an entity), so the method name should also reflect this.

Reply:
Agreed! I was actually meaning to change it to `CreateSensors` on all of them, so I'll do that instead. These plugins are so similar and there's so much duplication that there's value in reducing the diff so it's easier to compare them side-by-side.

I'll only address the `Create` functions in this PR not to scope creep, but I think we could do the same to `Add`, `Update` and `Remove`. In fact, I'd like to move a lot of this duplication to a templated helper class, but that's for the future :grimacing:

Reply:
75c78ad96d50849e61d03afb9e800a7f8daee6cd